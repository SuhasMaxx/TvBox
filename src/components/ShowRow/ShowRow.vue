<template src="./showrow.template.html"></template>
<script>
import ShowBlock from "../ShowBlock/ShowBlock.vue";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
/**
 * Row of shows with same Genre
 *
 * @class ShowRow
 */
export default {
    components: {
        ShowBlock,
        FontAwesomeIcon
    },
    props: {
        showListData: {
          type: Object,
          default: null
        }
    },
    data() {
      return {
        innerStyles: {},
        transitioning: false,
        step: '',
        disableNavBtns: false,
        showList: {}
      }
    },
    mounted(){
      this.showList = {...this.showListData};
      this.setStep();
      this.resetTranslate();
    },
    methods: {
      setStep (){
        const innerWidth = this.$refs.sliderRow.scrollWidth;
        this.step = `${innerWidth / 20 }px`;
        if(this.showList.showList.length < 10){
          this.disableNavBtns = true;
        }
      },
      scrollRight () {
        if (this.transitioning) return
        this.transitioning = true
        this.moveLeft()
        this.afterTransition(() => {
          const card = this.showList.showList.shift()
          this.showList.showList.push(card)
          this.resetTranslate()
          this.transitioning = false
        })
      },
      scrollLeft () {
        if (this.transitioning) return
        this.transitioning = true
        this.moveRight()
        this.afterTransition(() => {
          const card = this.showList.showList.pop();
          this.showList.showList.unshift(card)
          this.resetTranslate()
          this.transitioning = false
        })
      },
      moveLeft () {
        this.innerStyles = {
          transform: `translateX(-${this.step})
                      translateX(-${this.step})`
        }
      },
      moveRight () {
        this.innerStyles = {
          transform: `translateX(${this.step})
                      translateX(-${this.step})`
        }
      },
      afterTransition (callback) {
        const listener = () => {
          callback()
          this.$refs.sliderRow.removeEventListener('transitionend', listener)
        }
        this.$refs.sliderRow.addEventListener('transitionend', listener)
      },
      resetTranslate () {
        if(!this.disableNavBtns){
          this.innerStyles = {
            transition: 'none',
            transform: `translateX(-${this.step})`
          }
        }
        
      }
    }
}
</script>
<style src="./showrow.style.css" scoped></style>