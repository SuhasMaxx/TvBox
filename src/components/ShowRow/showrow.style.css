.genre-title{
    color: var(--cultured);
    margin-left: 1em;
    margin-bottom: 0.5em;
}

.show-row-container{
    display: flex;
    flex-wrap: nowrap;
    width: 100%;
    overflow-x: hidden;
}