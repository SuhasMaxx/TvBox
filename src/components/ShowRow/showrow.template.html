<div class="show-row-block">
    <h2 class="genre-title">{{showList.genre}}</h2>
    <div class="show-row-container">
        <div class="show-blocks-container" ref="sliderRow" :style="innerStyles">   
            <show-block v-for="show in showList.showList" :showDetails="show"></show-block>
        </div>
    </div>
    <div class="navigators" v-if="!disableNavBtns">
        <div class="showSlider left" @click="scrollLeft"><font-awesome-icon class="brand-icon" :icon="['fas', 'chevron-left']" /></div>
        <div class="showSlider right" @click="scrollRight"><font-awesome-icon class="brand-icon" :icon="['fas', 'chevron-right']" /></div>
    </div>
</div>