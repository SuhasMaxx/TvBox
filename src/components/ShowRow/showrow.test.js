import ShowRow from "./ShowRow.vue";
import { shallowMount } from "@vue/test-utils";
import showDetails from "../../utils/showDetails.mock.js";

describe("ShowRow.vue", () => {
  it("render the component", () => {
    const wrapper = shallowMount(ShowRow, {
      props: {
        showListData: showDetails.showRowData,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
