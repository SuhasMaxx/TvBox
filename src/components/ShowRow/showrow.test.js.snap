// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowRow.vue render the component 1`] = `
<div
  class="show-row-block"
>
  <h2
    class="genre-title"
  />
  <div
    class="show-row-container"
  >
    <div
      class="show-blocks-container"
    >
      
      
    </div>
  </div>
  <div
    class="navigators"
  >
    <div
      class="showSlider left"
    >
      <font-awesome-icon
        class="brand-icon"
        icon="fas,chevron-left"
      />
    </div>
    <div
      class="showSlider right"
    >
      <font-awesome-icon
        class="brand-icon"
        icon="fas,chevron-right"
      />
    </div>
  </div>
</div>
`;
